python/ray/tune/integration/torch.py

```diff
@@ -174,29 +224,31 @@ def DistributedTrainableCreator(
         analysis = tune.run(trainable_cls)
     """
     detect_checkpoint_function(func, abort=True)
+    if num_workers_per_host:
+        if num_workers % num_workers_per_host:
+            raise ValueError("`num_workers` must be an integer multiple "
+                             "of workers_per_node.")
 
     class WrappedDistributedTorchTrainable(_TorchTrainable):
         _function = func
         _num_workers = num_workers
-        _use_gpu = use_gpu
         _num_cpus_per_worker = num_cpus_per_worker
+        _num_gpus_per_worker = num_gpus_per_worker
+        _num_workers_per_host = num_workers_per_host
+        _timeout_s = timeout_s
 
         @classmethod
         def default_process_group_parameters(self) -> Dict:
             return dict(timeout=timedelta(timeout_s), backend=backend)
 
         @classmethod
         def default_resource_request(cls, config: Dict) -> Resources:
-            num_workers_ = int(config.get("num_workers", num_workers))
-            num_cpus = int(
-                config.get("num_cpus_per_worker", num_cpus_per_worker))
-            use_gpu_ = config.get("use_gpu", use_gpu)
 
             return Resources(
                 cpu=0,
                 gpu=0,
-                extra_cpu=num_cpus * num_workers_,
-                extra_gpu=num_workers_ if use_gpu_ else 0)
+                extra_cpu=num_cpus_per_worker * num_workers,
+                extra_gpu=num_gpus_per_worker * num_workers)
```

Comment:
ah I think it'll be important to keep the `use_gpu` parameter. We want:

1. if use_gpu = True, num_gpus_per_worker = None, then => num_gpus_per_worker = 1
2. if use_gpu = False, num_gpus_per_worker = None, then => num_gpus_per_worker = 0
3. if use_gpu = True, num_gpus_per_worker = 2, then => num_gpus_per_worker = 2
3. if use_gpu = False, num_gpus_per_worker = 2, then => num_gpus_per_worker = 2

Reply:
(same goes for TF)

Reply:
weird.. missed this. I wanted to get rid of this use_gpu flag to explicitly use num_gpu_per_worker